### PR TITLE
Refactored Benchmark flag and processing of command line arguments.

### PR DESCRIPTION
**Pull request by Sqeaky**
There is now more metadata, making it easier to make complex decisions in the future.

I added an Isolated Test Group which we should use for stable, short running tests, and seperated it from the the  Benchmark Test Group, which is for tests that can pass or fail purely performance measures.

### CODE REVIEW

File: src/ConsoleLogic.cpp [hunk 1]
```diff
@@ -102,13 +105,9 @@ namespace Mezzanine
...
-            // Replace with std::max_element once we get C++17
-            Mezzanine::String::size_type LongestName = 0;
-            for(const auto& Entry: TestGroups)
-            {
-                if(Entry.first.size()>LongestName)
-                    { LongestName=Entry.first.size(); }
-            }
+            Mezzanine::String::size_type LongestName = std::max_element(TestGroups.cbegin(),TestGroups.cend(),
+                [](const auto& Left, const auto& Right){ return Left.first.size() < Right.first.size(); }
+                )->first.size();
```
[the remark this answers is not in the document]
Comment (Sqeaky) [reply src/ConsoleLogic.cpp:110]: There were 3 of these loop to std::algo changes. All were suggested by CppCheck as I was grasping at straws for the msvc premature test termination. Now that all our compilers support c++17 I saw no reason not to use std::max_element

[the remark this answers is not in the document]
Comment (MakoEnergy) [reply src/ConsoleLogic.cpp:110]: It's not even that you are using std::max_element, although I suppose that is a small part of it.  It's just that it, plus a lambda, plus a dereference in a single statement isn't a trivial read with how it is laid out.

File: src/StringManipulation.cpp [hunk 2]
```diff
@@ -64,10 +66,14 @@ namespace Mezzanine
...
-            Results.reserve(StringToConvert.size());
-            for(const Mezzanine::StringView::value_type& Letter : StringToConvert)
-                { Results.push_back( static_cast<Mezzanine::String::value_type>(tolower(Letter)) ); }
+            using ViewChar = Mezzanine::StringView::value_type;
+            using StringChar = Mezzanine::String::value_type;
+
+            Mezzanine::String Results(StringToConvert);
+            std::transform(StringToConvert.begin(), StringToConvert.end(), Results.begin(),
+                           [](ViewChar Letter) -> StringChar { return static_cast<StringChar>(tolower(Letter)); }
+                           );
```
[the remark this answers is not in the document]
Comment (Sqeaky) [reply src/StringManipulation.cpp]: Another CppCheck inspired change. I need a destination with the correct size, and reserve allocation was not working. It gets the space, but the range isn't for iteration.

This is the one of the three I am least satisfied and I could see refactoring again, now that I figured out the problem I was having was all about reference expiry in the calling table and that is totally resolved now.

**Comment by Sqeaky** [comment]
I made the formatting better for each of the calls to an std algorithm and removed the double pass in that one part.

**Pull request merged**